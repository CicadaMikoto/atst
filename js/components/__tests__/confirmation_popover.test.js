import { createLocalVue, shallowMount } from '@vue/test-utils'
import VTooltip from 'v-tooltip'

import ConfirmationPopover from '../confirmation_popover'

const localVue = createLocalVue()
localVue.use(VTooltip)

describe('ConfirmationPopover', () => {
  const wrapper = shallowMount(ConfirmationPopover, {
    localVue,
    propsData: {
      action: '/some-url',
      btn_text: 'Do something dangerous',
      cancel_btn_text: 'Cancel',
      confirm_btn_text: 'Confirm',
      confirm_msg: 'Are you sure you want to do that?',
      csrf_token: '42',
    },
  })

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders form with hidden csrf input', () => {
    const input = wrapper.find('input[type=hidden]')
    expect(input.exists()).toBe(true)
    expect(input.attributes('value')).toBe('42')
  })
})
