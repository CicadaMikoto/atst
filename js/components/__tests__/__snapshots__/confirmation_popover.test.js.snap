// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationPopover matches snapshot 1`] = `
<v-popover-stub placement="top-start" delay="0" offset="0" trigger="click" container="body" popperoptions="[object Object]" popoverclass="vue-popover-theme" popoverbaseclass="tooltip popover" popoverinnerclass="tooltip-inner popover-inner" popoverwrapperclass="wrapper" popoverarrowclass="tooltip-arrow popover-arrow" autohide="true" handleresize="true"><template></template> <button type="button" class="tooltip-target">
    <div></div>
    Do something dangerous
  </button></v-popover-stub>
`;
